src/h5web/visualizations/line/ErrorBarCurve.tsx

```diff
@@ -23,58 +21,65 @@ function ErrorBarCurve(props: Props): ReactElement {
     ordinates,
     errors,
     color = DEFAULT_COLOR,
-    capLength = CAP_LENGTH,
     visible,
   } = props;
 
   const { abscissaScale, ordinateScale } = useCanvasScales();
 
-  const errorGeometry = useMemo(() => {
-    const points = ordinates.flatMap((val, index) => {
-      const [x, y] = [abscissaScale(abscissas[index]), ordinateScale(val)];
+  const [barsGeometry, capsGeometry] = useMemo(() => {
+    const capPoints: Vector2[] = [];
+    const barSegments: Vector2[] = [];
 
-      if (!Number.isFinite(y)) {
-        return [];
+    for (const index of ordinates.keys()) {
+      const val = ordinates[index];
+      const x = abscissaScale(abscissas[index]);
+      const y = ordinateScale(val);
+      if (!Number.isFinite(x) || !Number.isFinite(y)) {
+        continue;
       }
-
-      const error = errors[index];
+      const error = ordinateScale(errors[index]);
       if (error === 0) {
-        return [];
+        continue;
       }
-      const yerr_min = ordinateScale(val - error);
-      const yerr_max = ordinateScale(val + error);
 
-      return [
-        // First segment: bottom cap
-        ...lineSegment(
-          x - capLength / 2,
-          yerr_min,
-          x + capLength / 2,
-          yerr_min
-        ),
-        // Second segment: bottom error bar
-        ...lineSegment(x, yerr_min, x, y),
-        // Third segment: top error bar
-        ...lineSegment(x, y, x, yerr_max),
-        // Fourth segment: top cap
-        ...lineSegment(
-          x - capLength / 2,
-          yerr_max,
-          x + capLength / 2,
-          yerr_max
-        ),
-      ];
-    });
+      const bottom_cap = abscissaScale(val - error);
+      const top_cap = abscissaScale(val + error);
 
-    const geometry = new BufferGeometry();
-    geometry.setFromPoints(points);
-    return geometry;
-  }, [abscissaScale, abscissas, capLength, errors, ordinateScale, ordinates]);
+      const xyVector = new Vector2(x, y);
+
+      if (Number.isFinite(bottom_cap)) {
+        const bottomVector = new Vector2(x, bottom_cap);
+        capPoints.push(bottomVector);
+        barSegments.push(xyVector, bottomVector);
+      }
+
+      if (Number.isFinite(top_cap)) {
+        const topVector = new Vector2(x, top_cap);
+        capPoints.push(topVector);
+        barSegments.push(xyVector, topVector);
+      }
+    }
+
+    return [barSegments, capPoints].map((points) => {
+      const geometry = new BufferGeometry();
+      geometry.setFromPoints(points);
+      return geometry;
+    });
+  }, [abscissaScale, abscissas, errors, ordinateScale, ordinates]);
```

Comment:
I tried to improve in so many manners but cannot find a way out.

 The issue there is that I want to be able to treat `top` and `bottom` coordinates separately (to keep one even if the other is not finite) but at the same time, I need to separate `caps` and `bars` as they are used for separate geometries...